src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+
+</Wip>
+
+
+#### Caveats {/*caveats*/}
+* `'use server'` can only be used in server-side files; the resulting functions can be passed to Client Components through props.
 * Because the underlying network calls are always asynchronous, `'use server'` can be used only on async functions.
-* Directives like `'use server'` must be at the very beginning of their function or file, above any other code including imports (comments above directives are OK). They must be written with single or double quotes, not backticks. (The `'use xyz'` directive format somewhat resembles the `useXyz()` Hook naming convention, but the similarity is coincidental.)
+* server actions must be called in [transitions](). When used in a `<form action>`, form will ensure the server action is called in a transition. Elsewhere, you will need to manually call a server action in a transition.
+* Directives like `'use server'` must be at the very beginning of their function or file, above any other code including imports (comments above directives are OK). They must be written with single or double quotes, not backticks.
 
 ## Usage {/*usage*/}
 
-<Wip>
-This section is a work in progress. 
+### Server actions in forms {/*server-actions-in-forms*/}
+
+The most common use case of server actions will be calling server functions that mutate data. On the browser, the [HTML form element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) is the traditional approach for a user to submit a mutation. With React Server Components, React introduces first-class support for server actions in [forms](/reference/react-dom/components/form).
+
+Here is a form that allows a user to request a username.
+
+```js
+// UsernameForm.js
+'use client';
+
+export default UsernameForm({submit}) {
+  return (
+    <form action={submit}>
+      <input type="text" name="username" />
+      <button type="submit">Request</button>
+    </form>
+  );
+}
+```
+
+```js
+// App.js
+import UsernameForm from './UsernameForm';
+
+function requestUsername(formData) {
+  'use server';
+  const username = formData.get('username');
+  // ...
+}
+
+export default App() {
+  <UsernameForm submit={requestUsername}>
+}
+```
+
+In this example `UsernameForm` is a Client Component and receives a server action as the prop `submit`. When a user submits this form, there is a network request to the server function `requestUsername`.
+
+When calling a server action in a `<form action>`, React will supply the form's [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) as the first argument to the server action.
+
+<Note>
+
+We force `UsernameForm` as a Client Component with `'use client'` purely for demonstration. In general, it is not recommended to add the `'use client'` directive if the component does not use [client-only features](reference/react/use-client#usage).
```

Comment:
> In general, it is not recommended to add the `'use client'` directive if the component does not use [client-only features](reference/react/use-client#usage).

How do we feel about this sentence?

Do we want to make such a strong recommendation or does that feed the narrative that client components are bad? 

Could say something like "UsernameForm requires no client-only features, so this example would also work without "use client""

Alternatively, should we change the example to not need the note? Like, should we just remove use client? A form doesn't need that. Or if you want the use client example, should it use state or something?

Reply:
Ah, I see your next line builds on this being marked as a client component. So maybe use state somehow?

brainstorming an arbitrary idea:

```
export default UsernameForm({submit}) {
  const [username, setUsername] = useState('');
  return (
    <form action={submit}>
      <input type="text" name="username" value={username} onChange={(e) => setUsername(e.value)} />
      Username is {username.length} characters long!
      <button type="submit">Request</button>
    </form>
  );
}
```

Reply:
After seeing the other other example below, *shrug*, maybe the note is fine. lol. Maybe I've gone through the same thought process you did that landed you here 😅

Reply:
Yea after Sophie's review, I've removed the separation since I mistakenly thought progressive enhancement was affected by client vs. server rendering